src/ContainerTrait.php

```diff
@@ -14,22 +39,152 @@ public function _unique_element($desired = null)
         return $desired.($postfix > 1 ? ('_'.$postfix) : '');
     }
 
-    protected function _unique() {
+    /**
+     * Extension to add() method which will perform linking of
+     * the object with the current class.
+     */
+    protected function _add_Container($element, $args = [])
+    {
+        // Normalize the arguments, bring name out
+        if (is_string($args)) {
+
+            // passed as string
+            $args=[$args];
+        } elseif (!is_array($args)) {
+
+            throw new Exception(['Second argument must be array','arg2'=>$args]);
+        } elseif (is_array($args) && isset($args['name'])) {
+
+            // passed as ['name'=>'foo'];
+            $args[0]=$args['name'];
+            unset($args['name']);
+        } elseif (isset($element->short_name)) {
+
+            // element has a name already
+            $args[0]=$element->short_name;
```

Comment:
We do not want to mess about with short_name if it's set. 

I have to think of the implications here, because you can add object in multiple locations in theory. Let's leave it as-is for now until we have some realistic test-cases or use-cases.